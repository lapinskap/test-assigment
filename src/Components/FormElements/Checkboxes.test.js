import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Checkboxes, {
  CHECKBOXES_BUTTONS_DESELECT_ALL,
  CHECKBOXES_BUTTONS_SELECT_ALL,
} from './Checkboxes';

describe('<Autocomplete />', () => {
  const setState = jest.fn();
  const useStateSpy = jest.spyOn(React, 'useState');
  useStateSpy.mockImplementation((init) => [init, setState]);

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly with all props', () => {
    const div = document.createElement('div');
    div.setAttribute('id', 'form_tooltip_option_4');
    document.body.appendChild(div);

    const wrapper = renderer
      .create(<Checkboxes
        options={[
          { label: 'Option 1', value: 'option_1' },
          { label: 'Option 2', value: 'option_2' },
          { label: 'Option 3', value: 'option_3', required: true },
          { label: 'Option 4', value: 'option_4', tooltip: { content: 'Tooltip content' } },
          { label: 'Option 5', value: 'option_5' },
          { label: 'Option 6', value: 'option_6' },
          { label: 'Option 7', value: 'option_7' },
          { label: 'Option 8', value: 'option_8' },
          { label: 'Option 9', value: 'option_9' },
          { label: 'Option 10', value: 'option_10' },
          { label: 'Option 11', value: 'option_11' },
          { label: 'Option 12', value: 'option_12' },
        ]}
        disabled={false}
        afterLabel={<div>After label test</div>}
        label="Checkbox test"
        tooltip={<div>tooltip component</div>}
        columns={2}
        id="test_id"
        validation={['required']}
        value={['option_1']}
        errorMessage={'{"option_4":"Field is required"}'}
        onChange={() => {}}
        valueFormatter="numeric"
        inline={false}
        optionSwitcher={{
          onChange: () => {},
          disableIfNotChecked: true,
          checkedByDefault: true,
          label: 'ZmieÅ„',
        }}
      />);

    expect(wrapper.toJSON())
      .toMatchSnapshot();
  });

  it('renders correctly with only some props', () => {
    const wrapper = renderer
      .create(<Checkboxes
        options={[
          { label: 'Option 1', value: 'option_1' },
          { label: 'Option 2', value: 'option_2' },
          { label: 'Option 3', value: 'option_3' },
          { label: 'Option 4', value: 'option_4' },
          { label: 'Option 5', value: 'option_5' },
          { label: 'Option 6', value: 'option_6' },
          { label: 'Option 7', value: 'option_7' },
          { label: 'Option 8', value: 'option_8' },
          { label: 'Option 9', value: 'option_9' },
          { label: 'Option 10', value: 'option_10' },
          { label: 'Option 11', value: 'option_11' },
          { label: 'Option 12', value: 'option_12' },
        ]}
        disabled={false}
        label="Checkbox test"
        id="test_id"
        validation={['required']}
        value={['option_1']}
        onChange={() => {}}
        inline
      />);

    expect(wrapper.toJSON())
      .toMatchSnapshot();
  });

  it('correctly check option', () => {
    const onChangeSpy = jest.fn((id, values) => {
      expect(values.length).toBe(1);
      expect(values[0]).toBe('option_4');
    });

    const { container } = render(<Checkboxes
      options={[
        { label: 'Option 1', value: 'option_1' },
        { label: 'Option 2', value: 'option_2' },
        { label: 'Option 3', value: 'option_3' },
        { label: 'Option 4', value: 'option_4' },
        { label: 'Option 5', value: 'option_5' },
        { label: 'Option 6', value: 'option_6' },
        { label: 'Option 7', value: 'option_7' },
        { label: 'Option 8', value: 'option_8' },
        { label: 'Option 9', value: 'option_9' },
        { label: 'Option 10', value: 'option_10' },
        { label: 'Option 11', value: 'option_11' },
        { label: 'Option 12', value: 'option_12' },
      ]}
      valueFormatter="numeric"
      disabled={false}
      label="Checkbox test"
      id="test_id"
      value={null}
      validation={['required']}
      onChange={onChangeSpy}
    />);
    const option4 = container.querySelectorAll('input')[3];

    fireEvent.click(option4);
    expect(onChangeSpy).toHaveBeenCalledTimes(1);
  });

  it('correctly uncheck option', () => {
    const onChangeSpy = jest.fn((id, values) => {
      expect(values.length).toBe(0);
    });

    const { container } = render(<Checkboxes
      options={[
        { label: 'Option 1', value: 'option_1' },
        { label: 'Option 2', value: 'option_2' },
        { label: 'Option 3', value: 'option_3' },
        { label: 'Option 4', value: 'option_4' },
        { label: 'Option 5', value: 'option_5' },
        { label: 'Option 6', value: 'option_6' },
        { label: 'Option 7', value: 'option_7' },
        { label: 'Option 8', value: 'option_8' },
        { label: 'Option 9', value: 'option_9' },
        { label: 'Option 10', value: 'option_10' },
        { label: 'Option 11', value: 'option_11' },
        { label: 'Option 12', value: 'option_12' },
      ]}
      disabled={false}
      label="Checkbox test"
      id="test_id"
      validation={['required']}
      value={['option_4']}
      onChange={onChangeSpy}
    />);
    const option4 = container.querySelectorAll('input')[3];

    fireEvent.click(option4);
    expect(onChangeSpy).toHaveBeenCalledTimes(1);
  });

  it('correctly use option switcher', () => {
    const onSwitcherChangeSpy = jest.fn();

    const { container } = render(<Checkboxes
      options={[
        { label: 'Option 1', value: 'option_1' },
      ]}
      disabled={false}
      label="Checkbox test"
      id="test_id"
      validation={['required']}
      value={['option_4']}
      onChange={() => {}}
      optionSwitcher={{
        onChange: onSwitcherChangeSpy,
        disableIfChecked: true,
        checkedByDefault: true,
      }}
    />);
    const input = container.querySelectorAll('input')[0];
    const switcher = container.querySelectorAll('input')[1];
    expect(input.disabled).toBe(true);

    fireEvent.click(switcher);
    expect(input.disabled).toBe(false);
    fireEvent.click(switcher);
    expect(onSwitcherChangeSpy).toHaveBeenCalledTimes(2);
    expect(input.disabled).toBe(true);
  });

  it('correctly select all options by button', () => {
    const onChangeSpy = jest.fn((id, values) => {
      expect(values.length).toBe(3);
    });

    const { container } = render(<Checkboxes
      options={[
        { label: 'Option 1', value: 'option_1' },
        { label: 'Option 2', value: 'option_2' },
        { label: 'Option 3', value: 'option_3' },
      ]}
      label="Checkbox test"
      id="test_id"
      value={[]}
      onChange={onChangeSpy}
      buttons={CHECKBOXES_BUTTONS_SELECT_ALL}
    />);
    const button = container.querySelector('button');
    fireEvent.click(button);
    expect(onChangeSpy).toHaveBeenCalledTimes(1);
  }); it('correctly deselect all options by button', () => {
    const onChangeSpy = jest.fn((id, values) => {
      expect(values.length).toBe(0);
    });

    const { container } = render(<Checkboxes
      options={[
        { label: 'Option 1', value: 'option_1' },
        { label: 'Option 2', value: 'option_2' },
        { label: 'Option 3', value: 'option_3' },
      ]}
      label="Checkbox test"
      id="test_id"
      value={['option_1', 'option_2']}
      onChange={onChangeSpy}
      buttons={CHECKBOXES_BUTTONS_DESELECT_ALL}
    />);
    const button = container.querySelector('button');
    fireEvent.click(button);
    expect(onChangeSpy).toHaveBeenCalledTimes(1);
  });
});
