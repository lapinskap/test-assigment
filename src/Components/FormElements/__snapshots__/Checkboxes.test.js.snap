// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Autocomplete /> renders correctly with all props 1`] = `
Array [
  <label
    className=""
    data-t1="test_idLabel"
    htmlFor="test_id"
  >
    Checkbox test
     
    <div>
      tooltip component
    </div>
    <span
      className="ml-3"
      data-t1="afterLabel"
    >
      <div>
        After label test
      </div>
    </span>
  </label>,
  <div
    data-t1="test_id"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <div
          className="row"
          data-t1="test_id_option_1"
        >
          <div
            className="col-sm-2"
          >
            <div
              className=" is-invalid custom-checkbox custom-control"
              hidden={false}
            >
              <input
                aria-invalid={true}
                checked={true}
                className="is-invalid custom-control-input"
                data-t1="test_id"
                disabled={false}
                id="test_id_option_1"
                name="test_id_option_1"
                onChange={[Function]}
                type="checkbox"
                value="option_1"
              />
              <label
                className="custom-control-label"
                htmlFor="test_id_option_1"
              >
                Option 1
              </label>
            </div>
          </div>
          <div
            className="col-sm-auto"
          >
            <span
              className="my-2"
            >
              <div
                className="pt-2 custom-checkbox custom-control custom-control-inline"
                hidden={false}
              >
                <input
                  checked={true}
                  className="custom-control-input"
                  id="option_1_input_switch"
                  name="option_1_input_switch"
                  onChange={[Function]}
                  type="checkbox"
                />
                <label
                  className="custom-control-label"
                  htmlFor="option_1_input_switch"
                >
                  Zmień
                </label>
              </div>
            </span>
          </div>
        </div>
        <div
          className="row"
          data-t1="test_id_option_2"
        >
          <div
            className="col-sm-2"
          >
            <div
              className=" is-invalid custom-checkbox custom-control"
              hidden={false}
            >
              <input
                aria-invalid={true}
                checked={false}
                className="is-invalid custom-control-input"
                data-t1="test_id"
                disabled={false}
                id="test_id_option_2"
                name="test_id_option_2"
                onChange={[Function]}
                type="checkbox"
                value="option_2"
              />
              <label
                className="custom-control-label"
                htmlFor="test_id_option_2"
              >
                Option 2
              </label>
            </div>
          </div>
          <div
            className="col-sm-auto"
          >
            <span
              className="my-2"
            >
              <div
                className="pt-2 custom-checkbox custom-control custom-control-inline"
                hidden={false}
              >
                <input
                  checked={true}
                  className="custom-control-input"
                  id="option_2_input_switch"
                  name="option_2_input_switch"
                  onChange={[Function]}
                  type="checkbox"
                />
                <label
                  className="custom-control-label"
                  htmlFor="option_2_input_switch"
                >
                  Zmień
                </label>
              </div>
            </span>
          </div>
        </div>
        <div
          className="row"
          data-t1="test_id_option_3"
        >
          <div
            className="col-sm-2"
          >
            <div
              className=" is-invalid custom-checkbox custom-control"
              hidden={false}
            >
              <input
                aria-invalid={true}
                checked={false}
                className="is-invalid custom-control-input"
                data-t1="test_id"
                disabled={false}
                id="test_id_option_3"
                name="test_id_option_3"
                onChange={[Function]}
                type="checkbox"
                value="option_3"
              />
              <label
                className="custom-control-label"
                htmlFor="test_id_option_3"
              >
                Option 3
              </label>
            </div>
          </div>
          <div
            className="col-sm-auto"
          >
            <span
              className="my-2"
            >
              <div
                className="pt-2 custom-checkbox custom-control custom-control-inline"
                hidden={false}
              >
                <input
                  checked={true}
                  className="custom-control-input"
                  id="option_3_input_switch"
                  name="option_3_input_switch"
                  onChange={[Function]}
                  type="checkbox"
                />
                <label
                  className="custom-control-label"
                  htmlFor="option_3_input_switch"
                >
                  Zmień
                </label>
              </div>
            </span>
          </div>
        </div>
        <div
          className="row"
          data-t1="test_id_option_4"
        >
          <div
            className="col-sm-2"
          >
            <div
              className=" is-invalid custom-checkbox custom-control"
              hidden={false}
            >
              <input
                aria-invalid={true}
                checked={false}
                className="is-invalid custom-control-input"
                data-t1="test_id"
                disabled={false}
                id="test_id_option_4"
                name="test_id_option_4"
                onChange={[Function]}
                type="checkbox"
                value="option_4"
              />
              <label
                className="custom-control-label"
                htmlFor="test_id_option_4"
              >
                Option 4
                 
                <i
                  className="pe-7s-help1 text-primary font-weight-bold"
                  id="form_tooltip_option_4"
                />
              </label>
            </div>
          </div>
          <div
            className="col-sm-auto"
          >
            <span
              className="my-2"
            >
              <div
                className="pt-2 custom-checkbox custom-control custom-control-inline"
                hidden={false}
              >
                <input
                  checked={true}
                  className="custom-control-input"
                  id="option_4_input_switch"
                  name="option_4_input_switch"
                  onChange={[Function]}
                  type="checkbox"
                />
                <label
                  className="custom-control-label"
                  htmlFor="option_4_input_switch"
                >
                  Zmień
                </label>
              </div>
            </span>
          </div>
        </div>
        <div
          className="row"
          data-t1="test_id_option_5"
        >
          <div
            className="col-sm-2"
          >
            <div
              className=" is-invalid custom-checkbox custom-control"
              hidden={false}
            >
              <input
                aria-invalid={true}
                checked={false}
                className="is-invalid custom-control-input"
                data-t1="test_id"
                disabled={false}
                id="test_id_option_5"
                name="test_id_option_5"
                onChange={[Function]}
                type="checkbox"
                value="option_5"
              />
              <label
                className="custom-control-label"
                htmlFor="test_id_option_5"
              >
                Option 5
              </label>
            </div>
          </div>
          <div
            className="col-sm-auto"
          >
            <span
              className="my-2"
            >
              <div
                className="pt-2 custom-checkbox custom-control custom-control-inline"
                hidden={false}
              >
                <input
                  checked={true}
                  className="custom-control-input"
                  id="option_5_input_switch"
                  name="option_5_input_switch"
                  onChange={[Function]}
                  type="checkbox"
                />
                <label
                  className="custom-control-label"
                  htmlFor="option_5_input_switch"
                >
                  Zmień
                </label>
              </div>
            </span>
          </div>
        </div>
        <div
          className="row"
          data-t1="test_id_option_6"
        >
          <div
            className="col-sm-2"
          >
            <div
              className=" is-invalid custom-checkbox custom-control"
              hidden={false}
            >
              <input
                aria-invalid={true}
                checked={false}
                className="is-invalid custom-control-input"
                data-t1="test_id"
                disabled={false}
                id="test_id_option_6"
                name="test_id_option_6"
                onChange={[Function]}
                type="checkbox"
                value="option_6"
              />
              <label
                className="custom-control-label"
                htmlFor="test_id_option_6"
              >
                Option 6
              </label>
            </div>
          </div>
          <div
            className="col-sm-auto"
          >
            <span
              className="my-2"
            >
              <div
                className="pt-2 custom-checkbox custom-control custom-control-inline"
                hidden={false}
              >
                <input
                  checked={true}
                  className="custom-control-input"
                  id="option_6_input_switch"
                  name="option_6_input_switch"
                  onChange={[Function]}
                  type="checkbox"
                />
                <label
                  className="custom-control-label"
                  htmlFor="option_6_input_switch"
                >
                  Zmień
                </label>
              </div>
            </span>
          </div>
        </div>
      </div>
      <div
        className="col"
      >
        <div
          className="row"
          data-t1="test_id_option_7"
        >
          <div
            className="col-sm-2"
          >
            <div
              className=" is-invalid custom-checkbox custom-control"
              hidden={false}
            >
              <input
                aria-invalid={true}
                checked={false}
                className="is-invalid custom-control-input"
                data-t1="test_id"
                disabled={false}
                id="test_id_option_7"
                name="test_id_option_7"
                onChange={[Function]}
                type="checkbox"
                value="option_7"
              />
              <label
                className="custom-control-label"
                htmlFor="test_id_option_7"
              >
                Option 7
              </label>
            </div>
          </div>
          <div
            className="col-sm-auto"
          >
            <span
              className="my-2"
            >
              <div
                className="pt-2 custom-checkbox custom-control custom-control-inline"
                hidden={false}
              >
                <input
                  checked={true}
                  className="custom-control-input"
                  id="option_7_input_switch"
                  name="option_7_input_switch"
                  onChange={[Function]}
                  type="checkbox"
                />
                <label
                  className="custom-control-label"
                  htmlFor="option_7_input_switch"
                >
                  Zmień
                </label>
              </div>
            </span>
          </div>
        </div>
        <div
          className="row"
          data-t1="test_id_option_8"
        >
          <div
            className="col-sm-2"
          >
            <div
              className=" is-invalid custom-checkbox custom-control"
              hidden={false}
            >
              <input
                aria-invalid={true}
                checked={false}
                className="is-invalid custom-control-input"
                data-t1="test_id"
                disabled={false}
                id="test_id_option_8"
                name="test_id_option_8"
                onChange={[Function]}
                type="checkbox"
                value="option_8"
              />
              <label
                className="custom-control-label"
                htmlFor="test_id_option_8"
              >
                Option 8
              </label>
            </div>
          </div>
          <div
            className="col-sm-auto"
          >
            <span
              className="my-2"
            >
              <div
                className="pt-2 custom-checkbox custom-control custom-control-inline"
                hidden={false}
              >
                <input
                  checked={true}
                  className="custom-control-input"
                  id="option_8_input_switch"
                  name="option_8_input_switch"
                  onChange={[Function]}
                  type="checkbox"
                />
                <label
                  className="custom-control-label"
                  htmlFor="option_8_input_switch"
                >
                  Zmień
                </label>
              </div>
            </span>
          </div>
        </div>
        <div
          className="row"
          data-t1="test_id_option_9"
        >
          <div
            className="col-sm-2"
          >
            <div
              className=" is-invalid custom-checkbox custom-control"
              hidden={false}
            >
              <input
                aria-invalid={true}
                checked={false}
                className="is-invalid custom-control-input"
                data-t1="test_id"
                disabled={false}
                id="test_id_option_9"
                name="test_id_option_9"
                onChange={[Function]}
                type="checkbox"
                value="option_9"
              />
              <label
                className="custom-control-label"
                htmlFor="test_id_option_9"
              >
                Option 9
              </label>
            </div>
          </div>
          <div
            className="col-sm-auto"
          >
            <span
              className="my-2"
            >
              <div
                className="pt-2 custom-checkbox custom-control custom-control-inline"
                hidden={false}
              >
                <input
                  checked={true}
                  className="custom-control-input"
                  id="option_9_input_switch"
                  name="option_9_input_switch"
                  onChange={[Function]}
                  type="checkbox"
                />
                <label
                  className="custom-control-label"
                  htmlFor="option_9_input_switch"
                >
                  Zmień
                </label>
              </div>
            </span>
          </div>
        </div>
        <div
          className="row"
          data-t1="test_id_option_10"
        >
          <div
            className="col-sm-2"
          >
            <div
              className=" is-invalid custom-checkbox custom-control"
              hidden={false}
            >
              <input
                aria-invalid={true}
                checked={false}
                className="is-invalid custom-control-input"
                data-t1="test_id"
                disabled={false}
                id="test_id_option_10"
                name="test_id_option_10"
                onChange={[Function]}
                type="checkbox"
                value="option_10"
              />
              <label
                className="custom-control-label"
                htmlFor="test_id_option_10"
              >
                Option 10
              </label>
            </div>
          </div>
          <div
            className="col-sm-auto"
          >
            <span
              className="my-2"
            >
              <div
                className="pt-2 custom-checkbox custom-control custom-control-inline"
                hidden={false}
              >
                <input
                  checked={true}
                  className="custom-control-input"
                  id="option_10_input_switch"
                  name="option_10_input_switch"
                  onChange={[Function]}
                  type="checkbox"
                />
                <label
                  className="custom-control-label"
                  htmlFor="option_10_input_switch"
                >
                  Zmień
                </label>
              </div>
            </span>
          </div>
        </div>
        <div
          className="row"
          data-t1="test_id_option_11"
        >
          <div
            className="col-sm-2"
          >
            <div
              className=" is-invalid custom-checkbox custom-control"
              hidden={false}
            >
              <input
                aria-invalid={true}
                checked={false}
                className="is-invalid custom-control-input"
                data-t1="test_id"
                disabled={false}
                id="test_id_option_11"
                name="test_id_option_11"
                onChange={[Function]}
                type="checkbox"
                value="option_11"
              />
              <label
                className="custom-control-label"
                htmlFor="test_id_option_11"
              >
                Option 11
              </label>
            </div>
          </div>
          <div
            className="col-sm-auto"
          >
            <span
              className="my-2"
            >
              <div
                className="pt-2 custom-checkbox custom-control custom-control-inline"
                hidden={false}
              >
                <input
                  checked={true}
                  className="custom-control-input"
                  id="option_11_input_switch"
                  name="option_11_input_switch"
                  onChange={[Function]}
                  type="checkbox"
                />
                <label
                  className="custom-control-label"
                  htmlFor="option_11_input_switch"
                >
                  Zmień
                </label>
              </div>
            </span>
          </div>
        </div>
        <div
          className="row"
          data-t1="test_id_option_12"
        >
          <div
            className="col-sm-2"
          >
            <div
              className=" is-invalid custom-checkbox custom-control"
              hidden={false}
            >
              <input
                aria-invalid={true}
                checked={false}
                className="is-invalid custom-control-input"
                data-t1="test_id"
                disabled={false}
                id="test_id_option_12"
                name="test_id_option_12"
                onChange={[Function]}
                type="checkbox"
                value="option_12"
              />
              <label
                className="custom-control-label"
                htmlFor="test_id_option_12"
              >
                Option 12
              </label>
            </div>
          </div>
          <div
            className="col-sm-auto"
          >
            <span
              className="my-2"
            >
              <div
                className="pt-2 custom-checkbox custom-control custom-control-inline"
                hidden={false}
              >
                <input
                  checked={true}
                  className="custom-control-input"
                  id="option_12_input_switch"
                  name="option_12_input_switch"
                  onChange={[Function]}
                  type="checkbox"
                />
                <label
                  className="custom-control-label"
                  htmlFor="option_12_input_switch"
                >
                  Zmień
                </label>
              </div>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="invalid-feedback"
    style={
      Object {
        "display": "block",
      }
    }
  >
    {"option_4":"Field is required"}
  </div>,
]
`;

exports[`<Autocomplete /> renders correctly with only some props 1`] = `
Array [
  <label
    className=""
    data-t1="test_idLabel"
    htmlFor="test_id"
  >
    Checkbox test
     
  </label>,
  <div
    data-t1="test_id"
  >
    <div
      className="custom-checkbox custom-control custom-control-inline"
      hidden={false}
    >
      <input
        aria-invalid={false}
        checked={true}
        className="custom-control-input"
        data-t1="test_id"
        disabled={false}
        id="test_id_option_1"
        name="test_id_option_1"
        onChange={[Function]}
        type="checkbox"
        value="option_1"
      />
      <label
        className="custom-control-label"
        htmlFor="test_id_option_1"
      >
        Option 1
      </label>
    </div>
    <div
      className="custom-checkbox custom-control custom-control-inline"
      hidden={false}
    >
      <input
        aria-invalid={false}
        checked={false}
        className="custom-control-input"
        data-t1="test_id"
        disabled={false}
        id="test_id_option_2"
        name="test_id_option_2"
        onChange={[Function]}
        type="checkbox"
        value="option_2"
      />
      <label
        className="custom-control-label"
        htmlFor="test_id_option_2"
      >
        Option 2
      </label>
    </div>
    <div
      className="custom-checkbox custom-control custom-control-inline"
      hidden={false}
    >
      <input
        aria-invalid={false}
        checked={false}
        className="custom-control-input"
        data-t1="test_id"
        disabled={false}
        id="test_id_option_3"
        name="test_id_option_3"
        onChange={[Function]}
        type="checkbox"
        value="option_3"
      />
      <label
        className="custom-control-label"
        htmlFor="test_id_option_3"
      >
        Option 3
      </label>
    </div>
    <div
      className="custom-checkbox custom-control custom-control-inline"
      hidden={false}
    >
      <input
        aria-invalid={false}
        checked={false}
        className="custom-control-input"
        data-t1="test_id"
        disabled={false}
        id="test_id_option_4"
        name="test_id_option_4"
        onChange={[Function]}
        type="checkbox"
        value="option_4"
      />
      <label
        className="custom-control-label"
        htmlFor="test_id_option_4"
      >
        Option 4
      </label>
    </div>
    <div
      className="custom-checkbox custom-control custom-control-inline"
      hidden={false}
    >
      <input
        aria-invalid={false}
        checked={false}
        className="custom-control-input"
        data-t1="test_id"
        disabled={false}
        id="test_id_option_5"
        name="test_id_option_5"
        onChange={[Function]}
        type="checkbox"
        value="option_5"
      />
      <label
        className="custom-control-label"
        htmlFor="test_id_option_5"
      >
        Option 5
      </label>
    </div>
    <div
      className="custom-checkbox custom-control custom-control-inline"
      hidden={false}
    >
      <input
        aria-invalid={false}
        checked={false}
        className="custom-control-input"
        data-t1="test_id"
        disabled={false}
        id="test_id_option_6"
        name="test_id_option_6"
        onChange={[Function]}
        type="checkbox"
        value="option_6"
      />
      <label
        className="custom-control-label"
        htmlFor="test_id_option_6"
      >
        Option 6
      </label>
    </div>
    <div
      className="custom-checkbox custom-control custom-control-inline"
      hidden={false}
    >
      <input
        aria-invalid={false}
        checked={false}
        className="custom-control-input"
        data-t1="test_id"
        disabled={false}
        id="test_id_option_7"
        name="test_id_option_7"
        onChange={[Function]}
        type="checkbox"
        value="option_7"
      />
      <label
        className="custom-control-label"
        htmlFor="test_id_option_7"
      >
        Option 7
      </label>
    </div>
    <div
      className="custom-checkbox custom-control custom-control-inline"
      hidden={false}
    >
      <input
        aria-invalid={false}
        checked={false}
        className="custom-control-input"
        data-t1="test_id"
        disabled={false}
        id="test_id_option_8"
        name="test_id_option_8"
        onChange={[Function]}
        type="checkbox"
        value="option_8"
      />
      <label
        className="custom-control-label"
        htmlFor="test_id_option_8"
      >
        Option 8
      </label>
    </div>
    <div
      className="custom-checkbox custom-control custom-control-inline"
      hidden={false}
    >
      <input
        aria-invalid={false}
        checked={false}
        className="custom-control-input"
        data-t1="test_id"
        disabled={false}
        id="test_id_option_9"
        name="test_id_option_9"
        onChange={[Function]}
        type="checkbox"
        value="option_9"
      />
      <label
        className="custom-control-label"
        htmlFor="test_id_option_9"
      >
        Option 9
      </label>
    </div>
    <div
      className="custom-checkbox custom-control custom-control-inline"
      hidden={false}
    >
      <input
        aria-invalid={false}
        checked={false}
        className="custom-control-input"
        data-t1="test_id"
        disabled={false}
        id="test_id_option_10"
        name="test_id_option_10"
        onChange={[Function]}
        type="checkbox"
        value="option_10"
      />
      <label
        className="custom-control-label"
        htmlFor="test_id_option_10"
      >
        Option 10
      </label>
    </div>
    <div
      className="custom-checkbox custom-control custom-control-inline"
      hidden={false}
    >
      <input
        aria-invalid={false}
        checked={false}
        className="custom-control-input"
        data-t1="test_id"
        disabled={false}
        id="test_id_option_11"
        name="test_id_option_11"
        onChange={[Function]}
        type="checkbox"
        value="option_11"
      />
      <label
        className="custom-control-label"
        htmlFor="test_id_option_11"
      >
        Option 11
      </label>
    </div>
    <div
      className="custom-checkbox custom-control custom-control-inline"
      hidden={false}
    >
      <input
        aria-invalid={false}
        checked={false}
        className="custom-control-input"
        data-t1="test_id"
        disabled={false}
        id="test_id_option_12"
        name="test_id_option_12"
        onChange={[Function]}
        type="checkbox"
        value="option_12"
      />
      <label
        className="custom-control-label"
        htmlFor="test_id_option_12"
      >
        Option 12
      </label>
    </div>
  </div>,
]
`;
